Scripts/PAIF-N/02-install-nvidia-driver-vlcm.ps1

```
@@ -0,0 +1,186 @@
+}
+
+$uploadTasksId = $gpuParameters.NVIDIA | ForEach-Object {
+   # Upload the driver to vLCM
+   $SettingsDepotsOfflineCreateSpec = Initialize-SettingsDepotsOfflineCreateSpec `
+      -SourceType "PULL" `
+      -Location $_.Location `
+      -Description $_.Description
+
+   Invoke-CreateDepotsOfflineAsync -SettingsDepotsOfflineCreateSpec $SettingsDepotsOfflineCreateSpec
+}
+
+$uploadTasks = $uploadTasksId | ForEach-Object { Invoke-GetTask -task $_ }
+
+Write-Progress -Id 0 "Uploading NVIDIA vGPU driver into vLCM"
+$inProgress = $true
+while ($inProgress) {
+   Write-Verbose "Waiting for NVIDIA driver upload into vLCM"
+   $uploadTasks | ConvertTo-Json -Depth 5 | Write-Verbose
+
+   $subprocess = ''
+   $completed = 0
+   $total = 0
+
+   $inProgress = $false
+
+   foreach ($t in $uploadTasks) {
+      if ($t -and $t.status -ne 'SUCCEEDED' -and $t.status -ne 'FAILED') {
+         $inProgress = $true
+
+         if ($t.progress) {
+            if ($t.progress.message -and `
+               $t.progress.message.default_message) {
+               if ($subprocess.Length -gt 0) {
+                  $subprocess += ','
+               }
+               $subprocess += $t.progress.message.default_message
+            }
+            $completed += $t.progress.completed
+            $total += $t.progress.total
+         }
+      }
+   }
+
+   if ($total -eq 0) { $total = 100 }
+
+   Write-Progress -Id 0 "Uploading NVIDIA vGPU driver into vLCM" -Status $subprocess -PercentComplete (($completed * 100) / $total)
+
+   Start-Sleep -Seconds 1
+   $uploadTasks = $uploadTasksId | ForEach-Object { Invoke-GetTask -task $_ }
+}
+Write-Progress -Id 0 "Uploading NVIDIA vGPU driver into vLCM" -Completed
+
+# 3 vSphere LifeCycle Management Configuration
+$esxiBaseImage = Get-LcmImage `
+   -Type BaseImage `
+   -Version $gpuParameters.EsxiImageName
+$allComponents = Get-LcmImage -Type Component
+$components = $gpuParameters.NVIDIA | ForEach-Object {
+   $nvd = $_
+   $allComponents | `
```

Maybe fail if the components are not found in the repository?